src/staging.c

```diff
@@ -88,7 +88,7 @@ enum swupd_code do_staging(struct file *file, struct manifest *MoM)
 
 	/* make sure the directory where the file should be copied to exists
 	 * and is in deed a directory */
-	targetpath = sys_path_join("%s/%s", globals.path_prefix, rel_dir);
+	targetpath = sys_path_join("%s/%s", globals.path_prefix, dir);
```

Comment:
can't you just use `targetpath` in line 175 so you get rid or rel_dir entirely?
Like
```
string_or_die(&tarcommand, TAR_COMMAND " -C '%s' " TAR_PERM_ATTR_ARGS " -cf - './%s' 2> /dev/null | " TAR_COMMAND " -C '%s' " TAR_PERM_ATTR_ARGS " -xf - 2> /dev/null", rename_tmpdir, base, targetpath);
```

Reply:
good catch here. I haven't noticed that.